web-ui/src/main/resources/catalog/components/common/alert/AlertDirective.js

```diff
@@ -53,6 +64,20 @@
             'partials/alert.html',
         link: function(scope, element, attrs) {
           scope.alerts = gnAlertValue;
+
+          scope.closeAlert = function (alert) {
+            var alertPos = -1;
+            for(var i = 0; i < gnAlertValue.length; i++) {
```

Comment:
You can pass the current index using `$index` in the html template instead of doing a lookup in the array, see:
https://docs.angularjs.org/api/ng/directive/ngRepeat#overview

Reply:
Initially I implemented sending the index, but was not sure if could have some side effect when removing an alert. As long as the alerts are redisplayed again so new indexes are calculated, using the index should be fine I guess?

Reply:
Changed to use `$index` as tested and works fine. Also applied formatting to the javascript files changed in the pull request.